src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -442,19 +442,28 @@ def _binary_op(
         # pandas itself will ignore the axis argument when using Series.<op>.
         # Per default, it is set to axis=0. However, for the case of a Series interacting with
         # a DataFrame the behavior is axis=1. Manually check here for this case and adjust the axis.
+        lhs_series_rhs_dataframe = False
         if isinstance(self, pd.Series) and isinstance(other, pd.DataFrame):
-            axis = 1
+            lhs_series_rhs_dataframe = True
 
         new_query_compiler = self._query_compiler.binary_op(
             op=op,
             other=other,
-            axis=axis,
+            axis=1 if lhs_series_rhs_dataframe else axis,
             level=level,
             fill_value=fill_value,
             squeeze_self=squeeze_self,
             **kwargs,
         )
-        return self._create_or_update_from_compiler(new_query_compiler)
+
+        from snowflake.snowpark.modin.pandas.dataframe import DataFrame
+
+        # For a Series interacting with a DataFrame, always return a DataFrame
+        return (
+            DataFrame(query_compiler=new_query_compiler)
+            if lhs_series_rhs_dataframe
+            else self._create_or_update_from_compiler(new_query_compiler)
+        )
```

Comment:
This is actually a Modin bug -- See https://github.com/modin-project/modin/issues/7236

Reply:
can you add those into the comment too?

Reply:
Yep